astropy/io/fits/tests/test_core.py

```diff
@@ -650,7 +650,7 @@ def test_fits_file_handle_mode_combo(self):
 
     def test_open_from_url(self):
         import urllib.request
-        file_url = "file:///" + self.data('test0.fits')
+        file_url = 'file:///' +  self.data('test0.fits').lstrip('/')
```

Comment:
Belated thought: Maybe cleaner to use `pathlib` but since it works, let's move on. Thanks for wrapping this up, @bsipocz !